crates/rolldown_common/src/inner_bundler_options/types/transform_options.rs

```diff
@@ -24,31 +25,30 @@ pub enum JsxPreset {
   Preserve,
 }
 
+/// Trait for resolving tsconfig files. This abstraction allows `rolldown_common`
+/// to use tsconfig discovery without depending on `rolldown_resolver`.
+pub trait TsconfigResolver: Send + Sync + Debug {
+  fn find_tsconfig(
+    &self,
+    path: &Path,
+  ) -> Result<Option<Arc<oxc_resolver::TsConfig>>, oxc_resolver::ResolveError>;
+}
+
 /// Transform options with auto tsconfig discovery and caching
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct RawTransformOptions {
   pub base_options: Arc<BundlerTransformOptions>,
   /// Cache key: tsconfig path, or empty PathBuf for files without tsconfig
   pub cache: FxDashMap<PathBuf, Arc<OxcTransformOptions>>,
-  resolver: Arc<Resolver>,
+  pub resolver: Arc<dyn TsconfigResolver>,
```

Comment:
I wonder if it's better to use a generics here to reduce the indirection caused by dyn.